### PR TITLE
chore: remove bats test suite

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: test/bats/tests/azure_v1alpha1_secretproviderclass.yaml [hunk 1]
```diff
@@ -1,42 +0,0 @@
-apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
```
Comment (ritazh) [inline test/bats/tests/azure_v1alpha1_secretproviderclass.yaml:1]: Can you pls confirm we have an example of each of these spc test scenarios in the /examples/ folder?

Comment (aramase) [reply test/bats/tests/azure_v1alpha1_secretproviderclass.yaml:1]: Refactored the `examples/` folder to add more specific examples for configurations we support.

**Review by ritazh — APPROVED** [review]
lgtm

I don't think we are testing pv/pvc in the driver or provider anymore. should we remove these?

**Comment by aramase** [comment]
> I don't think we are testing pv/pvc in the driver or provider anymore. should we remove these?

Sounds good! I've removed it from the `examples` folder.

**Pull request merged**